ambari-server/src/main/java/org/apache/ambari/server/checks/KafkaPropertiesCheck.java

```diff
@@ -78,6 +79,14 @@ public KafkaPropertiesCheck() {
     return Lists.<CheckQualification> newArrayList(new KafkaPropertiesMinVersionQualification());
   }
 
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public Set<String> getApplicableServices(){
+    return Sets.newHashSet(KAFKA_SERVICE_NAME);
```

Comment:
`Collections.singleton` would be better, unless the result is modified by the caller.

Reply:
depends on what we wanna to achieve. The benefit from the `singleton` is that it creates immutable set from one element, while creating true immutable from several elements is more tricky until JDK 9: `Set.of(....)`

The context of usage, actually suppose the set to be immutable (not unmodifiable) and we can use guava for this, like: `return new ImmutableSet.Builder<String>().add(KAFKA_SERVICE_NAME).build();`

Currently it was done in the style of another checks, if you suppose that we need to formalize immutability of the elements, i can create jira for this change and update all pre-checks